gramps/gen/db/upgrade.py

```diff
@@ -52,12 +55,66 @@
     TAG_KEY,
 )
 from ..const import GRAMPS_LOCALE as glocale
+from .conversion_tools import convert_21
+from gramps.gen.lib.serialize import to_dict
 
 _ = glocale.translation.gettext
 
 LOG = logging.getLogger(".upgrade")
 
 
+def gramps_upgrade_21(self):
+    """
+    Add json_data field to tables.
+    """
+    length = 0
+    for key in self._get_table_func():
+        count_func = self._get_table_func(key, "count_func")
+        length += count_func()
+
+    self.set_total(length)
+
+    # First, do metadata:
+
+    self._txn_begin()
+    self.set_serializer("blob")
+    self.upgrade_table_for_json_data("metadata")
```

Comment:
```suggestion
    self.set_serializer("blob")
    self._txn_begin()
    # First, do metadata:
    self.upgrade_table_for_json_data("metadata")
```

Is there any reason to have the call to set_serializer inside the transaction?
Move the comment closer to the code. Comment may be superfluous, unless you want to expand on why you do the metadata first?

Reply:
> Is there any reason to have the call to set_serializer inside the transaction?

No, not really.

Reply:
I moved (and added) this line to the top of all of the conversion code. It might not be needed, but is a reminder to humans that these are all blob-based upgrades.